vendor/blockchain_contracts/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+#![warn(unused_extern_crates, missing_debug_implementations, rust_2018_idioms)]
+#![deny(unsafe_code)]
+
+#[macro_use]
+extern crate serde;
+
+use std::{cmp::Ordering, fmt::Display};
+
+pub mod ethereum;
+pub mod rfc003;
+
+#[derive(Debug, Eq)]
+pub struct Offset {
+    data: String,
+    start: usize,
+    end: usize,
+    length: usize,
+}
+
+impl Display for Offset {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(
+            f,
+            "{}\t{}\t\t{}\t{}",
+            self.data, self.start, self.end, self.length
+        )
+    }
+}
+
+pub fn format_table(mut offsets: Vec<Offset>) -> String {
+    let mut res = String::from("Data\t\t\t\tStart\tEnd\tLength");
+    offsets.sort_unstable();
+    for offset in offsets {
+        res = format!("{}\n{}", res, offset)
+    }
+    res
+}
+
+impl Offset {
+    fn new(data: String, start: usize, end: usize, length: usize) -> Offset {
+        Offset {
+            data,
+            start,
+            end,
+            length,
+        }
+    }
+}
+
+impl PartialEq for Offset {
+    fn eq(&self, other: &Offset) -> bool {
+        self.start == other.start
```

Comment:
:thinking: Why can you not just derive PartialEq if you need it? This implementations looks error prone. It is not used for anything critical but nevertheless, it should be correct.

Reply:
I want `Ord` so I need to derive `PartialEq`, `Eq` and `PartialOrd`.
Looks correct to me, please point out the in-correctness.

Reply:
But he's asking why you don't just `#[derive]` it.

Reply:
Because "Implementations of PartialEq, PartialOrd, and Ord must agree with each other."
https://doc.rust-lang.org/std/cmp/trait.Ord.html

Reply:
I think if you just make `start` the first field in the struct, you will get the desired behaviour through `#[derive]`.

Reply:
Now that I use an enum I need to manually implement some trait on the enum if I were to derive and put `start` as first field.
So I decided to implement the traits directly on the struct that needs to be `Ord`: `Offset`.